src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,241 @@
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
-
-    if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+    stars_ = deepcopy(stars)
+    if magnitude_cut:
+        try:
+            stars_ = stars_[stars_[band] < magnitude_cut]
+        except KeyError:
+            raise ValueError(
+                f"The requested catalogue has no compatible magnitude for the {band} band"
+            )
 
     if radius is not None:
-        if pointing is None:
+        if pointing:
+            stars_["separation"] = stars_["ra_dec"].separation(pointing)
+            stars_ = stars_[stars_["separation"] < radius]
+        else:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(
+    catalog: str | StarCatalog, magnitude_cutoff: float = 8.0, row_limit: int = 1000000
+) -> Table:
+    """
+    Utility function to download a star catalog for the get_bright_stars function.
+
+    Parameters
+    ----------
+    catalog : str or ctapipe.utils.astro.StarCatalog
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalog. Default: Yale.
+    magnitude_cutoff : float, optional
+        Maximum value for magnitude used in lookup. Default is 8.0.
+    row_limit : int, optional
+        Maximum number of rows for the star catalog lookup. Default is 1000000.
+
+    Returns
+    -------
+    astropy.table.Table
+        List of all stars after cuts with catalog numbers, magnitudes,
+        and coordinates as SkyCoord objects including proper motion.
+    """
+    from astroquery.vizier import Vizier
+
+    if isinstance(catalog, str):
+        catalog = StarCatalog[catalog]
+    catalog_info = catalog.value
+    catalog_name = catalog.name
+
+    vizier = Vizier(
+        catalog=catalog_info.directory,
+        columns=catalog_info.columns,
+        row_limit=row_limit,
+    )
+
+    stars = vizier.query_constraints(Vmag=f"<{magnitude_cutoff}")[0]
+
+    header = {
+        "ORIGIN": "CTAPIPE",
+        "JEPOCH": float(catalog_info.coordinates["epoch"].replace("J", "")),
+        "RADESYS": catalog_info.coordinates["frame"].upper(),
+        "MAGCUT": magnitude_cutoff,
+        "BAND": "V",
+        "CATALOG": catalog_name,
+        "REFERENC": catalog_info.directory,
+        "COLUMNS": "_".join(catalog_info.columns),
+    }
+
+    stars.meta = header
+
+    return stars
+
+
+def get_bright_stars(
+    time: Time,
+    catalog: StarCatalog | str = "Yale",
+    pointing: SkyCoord | None = None,
+    radius: Quantity | None = None,
+    magnitude_cut: float | None = None,
```

Comment:
This doesn't include and forward the `band` argument for the `select_stars`, should be added, right?